### PR TITLE
feat(Notion): add presence

### LINKED ISSUE #5231: Notion | notion.so
### Discussed in https://github.com/PreMiD/Presences/discussions/4538

<div type='discussions-op-text'>

<sup>Originally posted by **DecadeDecaf** January  5, 2021</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** DecadeDecaf#4128 

**What's the name of the service?** Notion 

**What should the Presence display?** The name of the page (and a way to distinguish whether or not the page is being read or edited). 

**If possible, please provide a logo for the service (512x512 minimum)** 
image 

**Important note:** I attempted adding Notion support, but my pull request was politely rejected. If you would like to see my code as a reference, you can do so [here](https://github.com/PreMiD/Presences/pull/2756). I fully understand why it was rejected, but I would still love to see Notion support on this plugin in the future! Thank you.</div>

### PR DESCRIPTION
**Pull request by kyrie25**
#### Resolves #5231 

<details>
<summary> Proof of the presence working: </summary>
<br>

image
image
image
image
image

</details>

### CODE REVIEW

**Comment by kyrie25** [comment]
Grabs overlay title as `.state`, otherwise return `document.title`
image

**Review by EncryptedDev — COMMENTED** [review]
What `overlayTitle`? Is it some sort of privacy filter?

**Comment by kyrie25** [comment]
> What `overlayTitle`? Is it some sort of privacy filter?

It's the title of the pop-up on the second screenshot but I had a brainfart and forgot what it's actually called 💀

**Comment by EncryptedDev** [comment]
> > What `overlayTitle`? Is it some sort of privacy filter?
> 
> It's the title of the pop-up on the second comment's screenshot but I had a brainfart and forgot what it's actually called 💀

In that case, I would add a privacy setting to prevent users stalking my document name

**Comment by kyrie25** [comment]
Gonna add it ASAP then

**Comment by kyrie25** [comment]
#### Added Settings tab for `Show Document Title`
<details>
<summary> Proof of the modification working: </summary>
<br>

kFomiAdmR0
Discord_9zbe9pOYFV
Discord_I3NZ1Brxkm
Discord_IAgBQ0THSk

</details>

File: websites/N/Notion/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "tags": [
+    "notion",
+    "notetaking",
+    "document",
+    "management"
+  ],
+  "settings": [
+    {
+      "id": "title",
+      "title": "Show Document Name",
```
Comment (EncryptedDev) [inline websites/N/Notion/dist/metadata.json:26]:
```suggestion
      "title": "Show document Nname",
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**